src/pages/RepoPage/BundlesTab/BundleContent/BundleSummary/BundleSelector.spec.tsx

```diff
@@ -0,0 +1,297 @@
+          name: 'bundle tab bundle selector',
+        })
+        expect(select).toBeInTheDocument()
+        expect(select).not.toBeDisabled()
+        expect(select).toHaveTextContent(/test-bundle/)
+      })
+    })
+
+    describe('navigating to a different bundle', () => {
+      describe('user selects a bundle', () => {
+        it('navigates to the selected bundle', async () => {
+          const { user } = setup({})
+          render(<BundleSelector />, { wrapper: wrapper() })
+
+          const select = await screen.findByRole('button', {
+            name: 'bundle tab bundle selector',
+          })
+          await user.click(select)
+
+          const bundle = await screen.findByText('bundle1')
+          await user.click(bundle)
+
+          await waitFor(() => {
+            expect(testLocation.pathname).toBe(
+              '/gh/codecov/test-repo/bundles/test-branch/bundle1'
+            )
+          })
+        })
+      })
+    })
+
+    describe('user searches for bundle', () => {
+      describe('branch is found', () => {
+        it('renders the found branch', async () => {
+          const { user } = setup({})
+          render(<BundleSelector />, {
+            wrapper: wrapper(),
+          })
+
+          const select = await screen.findByRole('button', {
+            name: 'bundle tab bundle selector',
+          })
+          await user.click(select)
+
+          const input = await screen.findByRole('combobox')
+          await user.type(input, 'bundle1')
+
+          const foundBundle = await screen.findByText('bundle1')
+          expect(foundBundle).toBeInTheDocument()
+        })
+      })
+
+      describe('branch is not found', () => {
+        it('displays no results found', async () => {
+          const { user } = setup({})
+          render(<BundleSelector />, {
+            wrapper: wrapper(),
+          })
+
+          const select = await screen.findByRole('button', {
+            name: 'bundle tab bundle selector',
+          })
+          await user.click(select)
+
+          const input = await screen.findByRole('combobox')
+          await user.type(input, 'searching for branch')
+
+          const notFound = await screen.findByText('No results found')
+          expect(notFound).toBeInTheDocument()
+        })
+      })
+
+      describe('user clears search', () => {
+        it('resets the select', async () => {
+          const { user } = setup({})
+          render(<BundleSelector />, {
+            wrapper: wrapper(),
+          })
+
+          const select = await screen.findByRole('button', {
+            name: 'bundle tab bundle selector',
+          })
+          await user.click(select)
+
+          const input = await screen.findByRole('combobox')
+          await user.type(input, 'test')
+
+          const notFound = await screen.findByText('No results found')
+          expect(notFound).toBeInTheDocument()
+
+          await user.type(input, '{backspace}{backspace}{backspace}{backspace}')
```

Comment:
This is cool. I didn't know you could do this

Reply:
https://testing-library.com/docs/react-testing-library/intro/

Reply:
there's some good stuff in there hah